src/nfa/nfa.cc

```diff
@@ -312,6 +312,136 @@ struct StackLevel {
     };
 };
 
+struct NodeData {
+    Post::const_iterator post_it;
+    Post::const_iterator post_end;
+    StateSet::const_iterator targets_it{};
+    StateSet::const_iterator targets_end{};
+    unsigned long index;
+    unsigned long lowlink;
+    bool initilized;
+    bool on_stack;
```

Comment:
Could we add a comment explaining what is the purpose of these member variables?